pose_teleoperation:
  virtual_world_frame_id:
    type: string
    default_value: "teleop_world"
    description: "Frame ID for the virtual world (reference frame for the 3D tracking controller)"
  controller_frame_id:
    type: string
    default_value: "LHR-FF29DD46"
    description: "Frame ID for the 3D tracking controller"
  base_frame_id:
    type: string
    default_value: "base_link"
    description: "Frame ID for the robot arm base"
  end_effector_frame_id:
    type: string
    default_value: "gripper_link"
    description: "Frame ID for the robot arm end effector"

  use_relative_pose:
    type: bool
    default_value: False
    description: "Use relative poses for teleoperation (if false, absolute poses are used)"

  joystick:
    calibrate_button:
      type: int
      default_value: 6
      description: "Button to calibrate the frame"
    teleoperate_button:
      type: int
      default_value: 7
      description: "Button to start/stop teleoperation (while being pressed)"
    home_button:
      type: int
      default_value: 8
      description: "Button to home the end effector"

  home:
    joint_controller_name:
      type: string
      default_value: "joint_trajectory_controller"
      description: "Name of the joint controller to use for homing"
    joint_names:
      type: string_array
      default_value: []
      description: "Names of the joints to home"
    joint_positions:
      type: double_array
      default_value: []
      description: "Positions of the joints to home"

  update_rate:
    type: int
    default_value: 100
    description: "Rate at which to publish target pose in Hz"

  cartesian_controller_name:
    type: string
    default_value: "cartesian_motion_controller"
    description: "Name of the Cartesian controller to use for teleoperation"

  bounds:
    enabled:
      type: bool
      default_value: false
      description: "Enable boundary limits for the end effector position based on the base frame"
    x_min:
      type: double
      default_value: -0.5
      description: "Minimum X position for the end effector in meters"
    x_max:
      type: double
      default_value: 0.5
      description: "Maximum X position for the end effector in meters"
    y_min:
      type: double
      default_value: -0.5
      description: "Minimum Y position for the end effector in meters"
    y_max:
      type: double
      default_value: 0.5
      description: "Maximum Y position for the end effector in meters"
    z_min:
      type: double
      default_value: 0.0
      description: "Minimum Z position for the end effector in meters"
    z_max:
      type: double
      default_value: 1.0
      description: "Maximum Z position for the end effector in meters"
